create/project_config.go

```diff
@@ -0,0 +1,61 @@
+package create
+
+import (
+	"context"
+	"time"
+
+	apps "github.com/ninech/apis/apps/v1alpha1"
+	"github.com/ninech/nctl/api"
+	"github.com/ninech/nctl/api/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+type configCmd struct {
+	Wait        bool              `default:"true" help:"Wait until API Project Config is fully created."`
+	WaitTimeout time.Duration     `default:"10s" help:"Duration to wait for API Project Config getting ready. Only relevant if wait is set."`
+	Size        string            `default:"micro" help:"Size of the app."`
+	Port        int32             `default:"8080" help:"Port the app is listening on."`
+	Replicas    int32             `default:"1" help:"Amount of replicas of the running app."`
+	Env         map[string]string `help:"Environment variables which are passed to the app at runtime."`
+}
+
+func (cmd *configCmd) Run(ctx context.Context, client *api.Client) error {
+	c := newCreator(client, cmd.newProjectConfig(client.Namespace), apps.ProjectConfigGroupKind)
+	ctx, cancel := context.WithTimeout(ctx, cmd.WaitTimeout)
+	defer cancel()
+
+	if err := c.createResource(ctx); err != nil {
+		return err
+	}
+
+	if !cmd.Wait {
+		return nil
+	}
+
+	return c.wait(ctx, waitStage{
+		objectList: &apps.ProjectConfigList{},
+		onResult:   resourceAvailable,
```

Comment:
I think this won't work for `ProjectConfigs` since we have no controller setting these resources to available. Essentially as soon as they are created they should be considered available as a new app will make use of them immediately.

Reply:
true 🤦